lib/td/client/api/job.rb

```diff
@@ -297,43 +313,28 @@ def job_result_download(job_id, format='msgpack', autodecode=true)
     current_total_chunk_size = 0
     infl = nil
     begin # LOOP of Network/Server errors
-      response = nil
-      client.get(url, params, header) do |res, chunk|
-        unless response
-          case res.status
-          when 200
-            if current_total_chunk_size != 0
-              # try to resume but the server returns 200
-              raise_error("Get job result failed", res)
-            end
-          when 206 # resuming
-          else
-            if res.status/100 == 5
-              raise HTTPServerException
-            end
-            raise_error("Get job result failed", res)
-          end
-          if infl.nil? && autodecode
-            case res.header['Content-Encoding'][0].to_s.downcase
-            when 'gzip'
-              infl = Zlib::Inflate.new(Zlib::MAX_WBITS + 16)
-            when 'deflate'
-              infl = Zlib::Inflate.new
-            end
+      response = client.get(url, params, header) do |res, chunk|
+        validate_response_status(res, current_total_chunk_size)
+        if infl.nil? && autodecode
+          case res.header['Content-Encoding'][0].to_s.downcase
+          when 'gzip'
+            infl = Zlib::Inflate.new(Zlib::MAX_WBITS + 16)
+          when 'deflate'
+            infl = Zlib::Inflate.new
           end
         end
-        response = res
         current_total_chunk_size += chunk.bytesize
         chunk = infl.inflate(chunk) if infl
         yield chunk, current_total_chunk_size
       end
+      validate_response_status(response)
```

Comment:
This line is intended for empty response body, so `client.get(...) do ... end` block wasn't called but `response` is assigned,  right?
I need some comments for that situation because looks like just `validate_response_status` is called twice.
Also I think `validate_response_status(response, 0) if response.body.nil?` would be more explicit.